File: uva_10905.py
import functools
def comp(a, b):
    if a+b > b+a:
        return -1
    else:
        return 1
while True:
    num=input()
    if num=='0':
        break
    input_data=[]
    data=input()
    data = data.split()
    for item in data:
        input_data.append(item)
    input_data=sorted(input_data, key=functools.cmp_to_key(comp))
    output=""
    for item in input_data:
        output+=item
    print(output)

File: uva_10905.cpp
# include <vector>
# include <string>
# include <iostream>
# include <algorithm>
using namespace std;
bool comp(string a, string b){
return (a+b) > (b+a);
}
int main(void){
    int num;
    while(cin >> num){
        if(!num)
            break;
        vector<string> input;
        for (int i=0; i<num; ++i){
            string item="";
            cin >> item;
            input.push_back(item);
        }
        sort(input.begin(), input.end(), comp);
        for (int i=0; i<input.size(); ++i)
            cout << input[i];
        cout << endl;
    }
    return 0;
}